src/Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -286,6 +289,8 @@ public override void Dispose()
 
             this.peerAddressManager.Dispose();
 
+            (this.partialValidator as PartialValidator)?.Dispose();
```

Comment:
Not sure about this line at all – why `?` And Also why `as`?

Reply:
Hmm this is in case validator would be overridden by something else.

Reply:
In which case you would like Idisposable on the interface because then you wouldn't need to know which particular implementation you have